sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -70,10 +76,30 @@ case class Mode(
     buffer
   }
 
-  override def eval(buffer: OpenHashMap[AnyRef, Long]): Any = {
-    if (buffer.isEmpty) {
+  override def eval(buff: OpenHashMap[AnyRef, Long]): Any = {
+    if (buff.isEmpty) {
       return null
     }
+    val buffer = if (isCollatedString(child)) {
+      val modeMap = buff.foldLeft(
```

Comment:
place `val collationId: Int = child.dataType.asInstanceOf[StringType].collationId` here so that avoid lazy.

Reply:
I am copying a pattern I have seen in other Expression implementations of collation. But I am fine making the suggested change.

Reply:
I think this is fine either way, I think we should generally aim to include `collationId` as part of the expression field (as we've done for other expressions), and there should be no harm since it's a lazy val

on the other hand, collationId is used only in one place here, but still I'd keep it as it is